java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -1127,4 +1127,25 @@ public void lookup_method_on_defered_type() throws Exception {
     //Lack of resolution when target type is deduced, we should be able to redo the lookup
     assertThat(reverse.usages()).isEmpty();
   }
+
+  @Test
+  public void infer_fully_lambda_types() {
+    Result result = Result.createFor("InferLambdaType");
+    LambdaExpressionTree lambda = ((LambdaExpressionTree) result.symbol("line").declaration().parent());
+    JavaType lambdaType = (JavaType) lambda.symbolType();
+    assertThat(lambdaType.isParameterized()).isTrue();
+    assertThat(lambdaType.is("java.util.Function"));
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2970)